Anixe.Ion/IIonReader.cs

```diff
@@ -2,17 +2,69 @@
     public interface IIonReader : IDisposable
     {
-        bool IsSectionHeader           { get; }
-        bool IsProperty                { get; }
-        bool IsComment                 { get; }
-        bool IsTableRow                { get; }
+        /// <summary>
+        /// Gets a value indicating whether this instance of IonReader is currently on section header.
+        /// </summary>
+        /// <value><c>true</c> if IonReader CurrentLine first character is equal to '['; otherwise, <c>false</c>.</value>
+        bool IsSectionHeader { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether this instance of IonReader is currently on property.
+        /// </summary>
+        /// <value><c>true</c> if other boolean properties are false; otherwise, <c>false</c>.</value>
+        bool IsProperty { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether this instance of IonReader is currently on comment.
+        /// </summary>
+        /// <value><c>true</c> if IonReader CurrentLine first character is equal to '#'; otherwise, <c>false</c>.</value>
+        bool IsComment { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether this instance of IonReader is currently on table row.
+        /// </summary>
+        /// <value><c>true</c> if IonReader CurrentLine first character is equal to '|'; otherwise, <c>false</c>.</value>
+        bool IsTableRow { get;}
+
+        /// <summary>
+        /// Gets a value indicating whether this instance of IonReader is currently on table header separator row. IMPORTANT: we recognize this 
+        /// property as a combination of first two characters as "|-". Fill your table rows without '-' character.
+        /// </summary>
+        /// <value><c>true</c> if IonReader CurrentLine first character is equal to '|-'; otherwise, <c>false</c>.</value>
         bool IsTableHeaderSeparatorRow { get; }
-        bool IsEmptyLine               { get; }
 
+        /// <summary>
+        /// Gets a value indicating whether a IonReader CurrentLine is null, empty, or consists only of white-space characters.
+        /// </summary>
+        /// <value><c>true</c> if string.IsNullOrWhiteSpace(CurrentLine); otherwise, <c>false</c>.</value>
+        bool IsEmptyLine { get; }
+
+        /// <summary>
+        /// Gets IonReader current line value.
+        /// </summary>
+        /// <value>The current line.</value>
         string CurrentLine { get; }
 
+        /// <summary>
+        /// Gets IonReader current section. It is changing only when CurrentLine is on section header.
+        /// </summary>
+        /// <value>The current section.</value>
+        string CurrentSection { get; }
+
+        /// <summary>
+        /// Gets IonReader current line numer.
+        /// </summary>
+        /// <value>The current line number.</value>
+        int CurrentLineNumber { get; }
+
+        /// <summary>
+        /// Reads the text line from stream.
```

Comment:
Reads the line from the stream

<returns><c>true</c> if read the line, otherwise <c>false</c></returns>